src/components/ui/ContributorLabel.tsx

```diff
@@ -0,0 +1,59 @@
+import Color from 'color'
+import {Contributor} from 'data/CONTRIBUTORS'
+import PropTypes from 'prop-types'
+import React, {CSSProperties, PureComponent} from 'react'
+import {Label} from 'semantic-ui-react'
+
+interface ContributorLabelProps {
+	contributor: string | Contributor
+	detail?: React.ReactNode
+}
+
+export default class ContributorLabel extends PureComponent<ContributorLabelProps> {
+	static propTypes = {
+		contributor: PropTypes.oneOfType([
+			PropTypes.string,
+			PropTypes.shape({
+				name: PropTypes.string.isRequired,
+				avatar: PropTypes.string,
+			}),
+		]).isRequired,
+		detail: PropTypes.node,
+	}
+
+	normalizeContributor = (contributor: string | Contributor): Contributor => {
+		if (typeof contributor === 'string') {
+			return {
+				name: contributor,
+				jobs: [],
+			}
```

Comment:
Is this not gonna need a stubbed avatar as well? I can't think of any reason contributor would be a string tho.

Reply:
No because `avatar` is optional and being checked in the render later if it even exists. As to why contributor would be a string... I have no idea, I was just refactoring what was there.